### PR TITLE
Add a `publish` subcommand to the xtask (which performs a dry-run by default)

### PR DESCRIPTION
**Pull request by jessebraham**
Seems to work as far as I can tell 😅

### CODE REVIEW

**Comment by jessebraham** [comment]
Considered adding a `publish` command instead which optionally allows for dry run, but I'm afraid I'll forget the argument. Can invert the logic I guess, but something like `cargo publish --publish` seems weird 😅  Open to suggestions here

**Comment by jessebraham** [comment]
Okay I ended up changing it to `publish`, which performs a dry-run by default but can optionally be supplied `--no-dry-run`. This command is a bit spicy but will be handy I guess 😅

File: xtask/src/main.rs [hunk 1]
```diff
@@ -770,6 +782,42 @@ fn lint_package(path: &Path, args: &[&str], fix: bool) -> Result<()> {
...
+    }
+
+    let builder = CargoArgsBuilder::default()
+        .subcommand("publish")
+        .args(&publish_args);
+
+    let args = builder.build();
+    log::debug!("{args:#?}");
+
+    // Execute `cargo publish --dry-run` from the package root:
```
Comment (SergioGasquez) [inline xtask/src/main.rs:815]:
```suggestion
    // Execute `cargo publish` command from the package root:
```

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks, just a nitpick suggestion that can totally be ignored

File: xtask/src/main.rs (hunk 1 above)
Comment (jessebraham) [reply xtask/src/main.rs:815]: Oops good catch, thanks

**Pull request merged**